File: program_data/github_java_sort_function/6/103.java
package com.paul.algorithm.sort;

import java.lang.reflect.Array;
import java.util.Arrays;
import java.util.LinkedList;

public class Radix {
  
  public int[] sort(int[] items) {
    if (items.length <= 1) {
      return items;
    }
    int[] result = Arrays.copyOf(items, items.length);
    boolean flag = true;
    int mod = 1;
    while (flag) {
      flag = false;
      mod *= 10;

      @SuppressWarnings("unchecked")
      LinkedList<Integer>[] bucket = (LinkedList<Integer>[])Array.newInstance(LinkedList.class,
          10);
      for (int i = 0; i < bucket.length; i++) {
        bucket[i] = new LinkedList<>();
      }

      for (int i = 0; i < result.length; i++) {
        int remainder = result[i] % mod;
        if (mod > 10) {
          remainder = remainder / (mod / 10);
        }

        bucket[remainder].add(result[i]);

        int num = result[i] / mod;
        if (num > 0) {
          flag = true;
        }
      }

      result = new int[result.length];
      int index = 0;
      for (int i = 0; i < bucket.length; i++) {
        for (int value : bucket[i]) {
          result[index] = value;
          index++;
        }
      }
    }

    return result;
  }
}

File: program_data/PKU_raw/87/103.c

int main(){
	int a[100],b[100],c[100],d[100],e[100],f[100];
	int i,j;
	for(i=0;i<100;i++){
		scanf("%d %d %d %d %d %d",&a[i],&b[i],&c[i],&d[i],&e[i],&f[i]);
		if(a[i]==0&&b[i]==0&&c[i]==0&&d[i]==0&&e[i]==0&&f[i]==0){
		j=i;
		break;}
	}     
	int k[100],m[100],n[100],result[100];
	for(i=0;i<j;i++){
		k[i]=d[i]+12;
		m[i]=a[i]*3600+b[i]*60+c[i];
		n[i]=k[i]*3600+e[i]*60+f[i];
		result[i]=n[i]-m[i];
		printf("%d\n",result[i]);
	}
	return 0;
}
